add-chain/addresses.go

```diff
@@ -15,154 +15,89 @@ type AddressData struct {
 	Address string `json:"address"`
 }
 
-var (
-	// Addresses to retrieve from JSON
-	AddressManager                    = "AddressManager"
-	L1CrossDomainMessengerProxy       = "L1CrossDomainMessengerProxy"
-	L1ERC721BridgeProxy               = "L1ERC721BridgeProxy"
-	L1StandardBridgeProxy             = "L1StandardBridgeProxy"
-	OptimismMintableERC20FactoryProxy = "OptimismMintableERC20FactoryProxy"
-	SystemConfigProxy                 = "SystemConfigProxy"
-	OptimismPortalProxy               = "OptimismPortalProxy"
-	ProxyAdmin                        = "ProxyAdmin"
-
-	// Addresses to retrieve from chain
-	SuperchainConfig  = "SuperchainConfig"
-	Guardian          = "Guardian"
-	Challenger        = "Challenger"
-	ProxyAdminOwner   = "ProxyAdminOwner"
-	SystemConfigOwner = "SystemConfigOwner"
-	Proposer          = "Proposer"
-	UnsafeBlockSigner = "UnsafeBlockSigner"
-	BatchSubmitter    = "BatchSubmitter"
-
-	// Non Fault Proof contracts
-	L2OutputOracleProxy = "L2OutputOracleProxy"
-
-	// Fault Proof contracts:
-	AnchorStateRegistryProxy = "AnchorStateRegistryProxy"
-	DelayedWETHProxy         = "DelayedWETHProxy"
-	DisputeGameFactoryProxy  = "DisputeGameFactoryProxy"
-	FaultDisputeGame         = "FaultDisputeGame"
-	MIPS                     = "MIPS"
-	PermissionedDisputeGame  = "PermissionedDisputeGame"
-	PreimageOracle           = "PreimageOracle"
-)
-
-func readAddressesFromChain(addresses map[string]string, l1RpcUrl string, isFaultProofs bool) error {
+func readAddressesFromChain(addresses *superchain.AddressList, l1RpcUrl string, isFaultProofs bool) error {
 	// SuperchainConfig
-	address, err := castCall(addresses[OptimismPortalProxy], "superchainConfig()(address)", l1RpcUrl)
+	address, err := castCall(addresses.OptimismPortalProxy.String(), "superchainConfig()(address)", l1RpcUrl)
```

Comment:
Many `castCall` calls seem to now have an `Address` at hand, instead of the address `string`. If there aren't many invocations left to `castCall` that pass a string instead of an `Address`, maybe a good opportunity to clean it up and let `castCall`'s first argument be of type `Address`. It would be more safer anyways, since the argument _must_ be an address.

Reply:
Can also do this in a follow-up

Reply:
https://github.com/ethereum-optimism/superchain-registry/pull/425/commits/df412f3da8785483b5dabb20388190bc0663c376